src/client/ducks/bfstats/selectors/index.js

```diff
@@ -5,25 +5,36 @@ import _ from 'lodash'
 export type { StatusType } from 'ducks/fetch/selectors'
 
 export const getBuildsEntities = (state: Object) =>
-  _.get(state, ['entities', 'builds'], {})
+  _.get(state, ['entities', 'builds'], null)
 
 export const getSourceStampsEntities = (state: Object) =>
-  _.get(state, ['entities', 'sourceStamps'], {})
+  _.get(state, ['entities', 'sourceStamps'], null)
 
 export const getRevision = (state: Object, props: Object) =>
-  (props ? props.revision : null)
+  (props && props.revision ? props.revision : null)
 
 export const getSourceStamp = createSelector(
   getSourceStampsEntities, getRevision,
-  (sourceStamps, revision) => (sourceStamps && revision ? sourceStamps[revision] : null)
+  (sourceStamps, revision) => (
+    sourceStamps && revision && sourceStamps.hasOwnProperty(revision)
+      ? sourceStamps[revision] : null)
 )
 
 export const getBuilds = createSelector(
   getBuildsEntities, getSourceStamp,
   (builds, sourceStamp) => {
     if (!builds || !sourceStamp) {
-      return []
+      return null
     }
-    return sourceStamp.nodes.map(id => builds[id])
+
+    return sourceStamp.latestBuilds.nodes.map(id => builds[id])
+  }
+)
+
+export const getABVBuild = createSelector(
+  getBuilds,
+  builds => {
+    const index = _.findIndex(builds, build => build.builder.name === 'proj0-ABuildVerification')
```

Comment:
I think it would be nicer if we rather than filter based on them being top-level builds (i.e. they're the root of the build chain). The latestBuilds property does not have the possibility to filter based on that, but if changing the GraphQL query to do: `builds(rootOnly: true) { ... }` we would be able to show all top-level builds.
This can be very useful when people are doing graphics-only changes that we're reasonably certain are constrained to graphics code so they can settle with only running stable graphics tests. Then release management can decide to let it pass without a full ABV (which might not necessarily test that code at all, delaying discovery until QV).